awx/main/fields.py

```diff
@@ -686,11 +686,10 @@ def schema(self, model_instance):
             'properties': {
                 'file': {
                     'type': 'object',
-                    'properties': {
-                        'template': {'type': 'string'},
+                    'patternProperties': {
+                        '^template(\.[a-zA-Z_]+)?$': {'type': 'string'},
```

Comment:
Technically speaking, this _could be_ any valid jinja variable name.  Instead of:

```regex
^template(\.[a-zA-Z_]+)?$
```

...you probably want to match: https://github.com/ansible/awx/blob/96904968d850c4af22e58b69c0aa3a3221d70e9c/awx/main/fields.py#L615

```regex
^template(\.[a-zA-Z_]+[a-zA-Z0-9_]*)?$
```

Disclaimer: I _think_ that is right :)

Reply:
@ryanpetrello - is the `template.*` more of a convention instead of a requirement, then? It seemed like the `template.*` syntax was a requirement. (I definitely have it hard-coded elsewhere)

Reply:
Since I'm setting the schema for the `file` field, specifically, it seemed worth enforcing the syntax here. It sounds like I might be missing something though.

Reply:
I'm saying that it's possible for people to use jinja2 variables in their injectors that contain more complicated strings than those matched by `[a-zA-Z_]+`

What if you had an injectors like this?

```
"injectors": {
    "file": {
        "template.cert_1": "{{cert}}",
        "template.key_1": "{{key}}"
    },
    "env": {
        "MY_CERT": "{{tower.filename.cert_1}}",
        "MY_KEY": "{{tower.filename.key_1}}"
    }
}
```